### PR TITLE
Osama bazbooz-week1-APIs

### PR DESCRIPTION
**Pull request by obazbooz**

### CODE REVIEW

**Comment by jontra000** [comment]
Hi @obazbooz, good work on exercises 1, 3 and 5. These all look good! There are some issues with exercises 2 and 4 however.

For ex2, check the results you get when running with 10 or 99 as an input. It is always a good idea to test our code with the edge cases, in this case the values at the edge of the range we are interested in (10 - 99).

For ex4, the `rollTheDices` function returns `Resolved! [ undefined, undefined, undefined, undefined, undefined ]`, not the result we want. I suggest you walk through `rollTheDices` with the debugger and find where the problem is.

Lastly, your explanation in ex4 is correct, but not the complete explanation. Once you've fixed the problem with `rollTheDices`, try adding the return statement as you suggested and see what happens.

File: 3-UsingAPIs/Week1/homework/ex2-checkDoubleDigits.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 'use strict';
+
+const { error } = require('winston');
```
Comment (jontra000) [inline 3-UsingAPIs/Week1/homework/ex2-checkDoubleDigits.js:3]: Make sure to delete any unneeded code, this keeps it clean and tidy, and easier for your co-workers to understand.

**Comment by obazbooz** [comment]
Dear @jontra000,

Thanks for your feedback.
I have worked on the issues and points you have mentioned and I did the following:

1. I corrected the if condition statement in exercise 2 and also I commented out the statement in line 3 because I did not use it. But it was already in the starting code of the HW. I don't know if I have to use it some how or not!
2. I corrected the issue with getting undefined values with the resolved situation in ex4, by adding a return in side the mapping method.
3. I tried as you suggested in ex4 to try my solution by adding return in the catch but I was still getting some results after printing a rejected statement on the screen and I guess that because of the asynchronous statements in the event loop and to solve that I have used the setTimeout method.
Thanks again for you time and I will be happy to get your feedback about my changes.

Best regards,
Osama

**Review by jontra000 — APPROVED** [review]
Nice work, all looks good now. Just a couple of small bits of advice.

Re: the require statement in exercise 2: This line is not in the original homework. It might have been added automatically by Visual Studio because it thought you needed it. It's good to keep an eye on this sort of thing because it can interfere with how your code behaves.

Re: the explanation in exercise 4: Your explanation is correct, but FYI it is not necessary to add the timeouts. It's OK, and in fact preferable, that our promise returns and lets us know the result as soon as it can when one of the dice falls on the floor. Afterwards some dice might still be bouncing around and printing their values, but in this example we are OK with this.

**Comment by obazbooz** [comment]
Thanks @jontra000 for your advice and feedback,I will pay attention for that.
